lambda-cloudformation-custom-resource/src/test/scala/feral/lambda/cloudformation/CloudFormationCustomResourceArbitraries.scala

```diff
@@ -0,0 +1,201 @@
+
+  val genRequestId: Gen[RequestId] = arbitrary[UUID].map(_.toString).map(RequestId(_))
+  implicit val arbRequestId: Arbitrary[RequestId] = Arbitrary(genRequestId)
+
+  val genResourceType: Gen[ResourceType] =
+    for {
+      // max of 60 alphanumeric and _@- characters, see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/crpg-ref-requests.html
+      length <- Gen.chooseNum(1, 60)
+      str <- Gen.stringOfN(
+        length,
+        Gen.oneOf(Gen.alphaNumChar, Gen.const('_'), Gen.const('@'), Gen.const('-')))
+    } yield ResourceType(str)
+  implicit val arbResourceType: Arbitrary[ResourceType] = Arbitrary(genResourceType)
+
+  val genLogicalResourceId: Gen[LogicalResourceId] = arbitrary[String].map(LogicalResourceId(_))
+  implicit val arbLogicalResourceId: Arbitrary[LogicalResourceId] = Arbitrary(
+    genLogicalResourceId)
+
+  val genPhysicalResourceId: Gen[PhysicalResourceId] =
+    for {
+      length <- Gen.chooseNum(1, 1024)
+      str <- Gen.stringOfN(length, Gen.asciiPrintableChar).map(trimStringToByteLength(1024)(_))
+    } yield PhysicalResourceId.unsafeApply(str)
+  implicit val arbPhysicalResourceId: Arbitrary[PhysicalResourceId] = Arbitrary(
+    genPhysicalResourceId)
+
+  @tailrec
+  private def trimStringToByteLength(
+      length: Int)(s: String, encoding: Charset = Charset.forName("UTF-8")): String =
+    if (s.getBytes(encoding).length > length)
+      trimStringToByteLength(length - 1)(s.drop(1), encoding)
+    else s
+
+  def genCloudFormationCustomResourceRequest[A: Arbitrary]
+      : Gen[CloudFormationCustomResourceRequest[A]] =
+    for {
+      requestType <- arbitrary[CloudFormationRequestType]
+      responseURL <- arbitrary[Uri]
+      stackId <- arbitrary[StackId]
+      requestId <- arbitrary[RequestId]
+      resourceType <- arbitrary[ResourceType]
+      logicalResourceId <- arbitrary[LogicalResourceId]
+      physicalResourceId <- arbitrary[Option[PhysicalResourceId]]
+      resourceProperties <- arbitrary[A]
+      oldResourceProperties <- arbitrary[Option[JsonObject]]
+    } yield CloudFormationCustomResourceRequest(
+      requestType,
+      responseURL,
+      stackId,
+      requestId,
+      resourceType,
+      logicalResourceId,
+      physicalResourceId,
+      resourceProperties,
+      oldResourceProperties
+    )
+
+  implicit def arbCloudFormationCustomResourceRequest[A: Arbitrary]
+      : Arbitrary[CloudFormationCustomResourceRequest[A]] = Arbitrary(
+    genCloudFormationCustomResourceRequest[A])
+
+  def genLambdaEnv[F[_]: Applicative, A: Arbitrary]
+      : Gen[LambdaEnv[F, CloudFormationCustomResourceRequest[A]]] =
+    for {
+      e <- arbitrary[CloudFormationCustomResourceRequest[A]]
+      c <- arbitrary[Context[F]]
+    } yield new LambdaEnv[F, CloudFormationCustomResourceRequest[A]] {
+      override def event: F[CloudFormationCustomResourceRequest[A]] = e.pure[F]
+      override def context: F[Context[F]] = c.pure[F]
+    }
```

Comment:
Instead of unsealing `LambdaEnv` can we just upstream this constructor? I'd considered adding it before, but hesitated due to https://github.com/typelevel/feral/issues/85. We can keep it package-private.

Reply:
Yep, good idea